service/controller/chart/v1/resource/release/delete.go

```diff
@@ -24,46 +22,25 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 
 		err = r.helmClient.DeleteRelease(ctx, releaseState.Name, helm.DeletePurge(true))
 		if helmclient.IsReleaseNotFound(err) {
-			// Fall through.
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q already deleted", releaseState.Name))
 			return nil
 		} else if err != nil {
 			return microerror.Mask(err)
 		}
 
-		var rel *helmclient.ReleaseContent
-		{
-			o := func() error {
-				rel, err = r.helmClient.GetReleaseContent(ctx, releaseState.Name)
-				if rel != nil {
-					return microerror.Maskf(waitError, "release %#q still exists", releaseState.Name)
-				} else if helmclient.IsReleaseNotFound(err) {
-					// Fall through.
-					return nil
-				} else if err != nil {
-					return microerror.Mask(err)
-				}
-
-				return nil
-			}
-			b := backoff.NewMaxRetries(3, 5*time.Second)
-			n := backoff.NewNotifier(r.logger, ctx)
-
-			err := backoff.RetryNotify(o, b, n)
-			if IsWait(err) {
-				// We timed out and the helm release still exists. We cancel the
-				// resource and keep the finalizer. We will retry the delete in
-				// the next reconciliation loop.
-				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to delete release %#q", releaseState.Name))
-				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-				resourcecanceledcontext.SetCanceled(ctx)
-				return nil
-			} else if err != nil {
-				return microerror.Mask(err)
-			}
-
+		rel, err := r.helmClient.GetReleaseContent(ctx, releaseState.Name)
+		if rel != nil {
+			// Release still exists. We cancel the resource and keep the finalizer.
+			// We will retry the delete in the next reconciliation loop.
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q still exists", releaseState.Name))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
Where do we keep it though? The call to keep it is missing.

Reply:
Ack. More ☕️ needed. Thx I get it sorted.

Reply:
RTFM 🤦‍♂ https://github.com/giantswarm/operatorkit/blob/5dc93b8ef81d2684b755a813327aada03d7dc2f8/docs/using_finalizers.md#control-flow

@xh3b4sd PTAL

Reply:
```suggestion
			// Release still exists. We cancel the resource and keep the finalizer.
			// We will retry the delete in the next reconciliation loop.
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q still exists", releaseState.Name))
			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
			resourcecanceledcontext.SetCanceled(ctx)
			r.logger.LogCtx(ctx, "level", "debug", "message", "keeping finalizers")
```